traitsui/tests/editors/test_enum_editor.py

```diff
@@ -101,18 +94,237 @@ def set_combobox_index(combobox, idx):
         combobox.setCurrentIndex(idx)
 
 
-class TestEnumEditor(unittest.TestCase):
-    def check_enum_text_update(self, view):
+def finish_combobox_text_entry(combobox):
+    """ Finish text entry given combobox. """
+    if is_current_backend_wx():
+        import wx
+
+        event = wx.CommandEvent(wx.EVT_TEXT_ENTER.typeId, combobox.GetId())
+        wx.PostEvent(combobox, event)
+
+    elif is_current_backend_qt4():
+        combobox.lineEdit().editingFinished.emit()
+
```

Comment:
Will add an `else` branch 👍  Although I don't see why tests for existing toolkits would start failing even if `else` branch isn't added -- am I missing something?

As for the new pattern, to me adding an `elif` branch to a helper method seems easier than going through this framework -- there is only one docstring and one signature to maintain and adding 
```
elif new_toolkit:
    implementation
```
seems easier than
```
@function.register("new_toolkit")
def function(signature):
    implementation
```

Reply:
> Although I don't see why tests for existing toolkits would start failing even if else branch isn't added -- am I missing something?

Sorry for my weird way of putting things. 

If the else branch isn't added, and if we have a new toolkit, when we run the test suite with this new toolkit, the tests here will not be skipped (because the toolkit isn't null). Since `is_current_backend_wx` and `is_current_backend_qt4` will return false, the tests will likely fail because the helper functions are no-op in that new toolkit.

Reply:
> there is only one docstring and one signature to maintain and adding

I would probably only maintain the docstring once only like this:

```
@toolkit_dispatch
def get_something(widget):
    """ Docstring here """

@get_something("wx")
def _(widget):
    return widget.GetName()
```
This is similar to the `functools.singledispatch`: It is a functional way to achieve polymorphism.